utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -208,9 +226,138 @@ class PerformanceMetricsAssessorImplTest {
 
   @Test
   @RunParameterized(
-    OppiaParameterizedTestRunner.Iteration("memoryEqualToLowerBound", "totalMemory=0"),
-    OppiaParameterizedTestRunner.Iteration("memoryInRange", "totalMemory=1147483648"),
-    OppiaParameterizedTestRunner.Iteration("memoryJustBelowUpperBound", "totalMemory=2147483647")
+    Iteration(
+      "zeroValue",
+      "secondCpuValue=1000",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=2",
+      "relativeCpuUsage=0.00"
+    ),
+    Iteration(
+      "prodValueExample1",
+      "secondCpuValue=1100",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=2",
+      "relativeCpuUsage=0.50"
+    ),
+    Iteration(
+      "prodValueExample2",
+      "secondCpuValue=12100",
+      "secondAppTimeValue=1869790700",
+      "secondNumberOfOnlineCoresValue=8",
+      "relativeCpuUsage=0.0000077"
+    ),
+    Iteration(
+      "prodValueExample3",
+      "secondCpuValue=2100",
+      "secondAppTimeValue=186933790700",
+      "secondNumberOfOnlineCoresValue=3",
+      "relativeCpuUsage=0.00000000239"
+    ),
+  )
+  fun testAssessor_setFirstAndSecondSnapshot_verifyCorrectCpuUsageIsReturned() {
+    val firstSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      TEST_FIRST_PROCESS_TIME,
+      TEST_FIRST_CPU_TIME,
+      TEST_FIRST_NUMBER_OF_CORES
+    )
+    val secondSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      secondAppTimeValue,
+      secondCpuValue,
+      secondNumberOfOnlineCoresValue
+    )
+    val returnedRelativeCpuUsage =
+      performanceMetricsAssessorImpl.getRelativeCpuUsage(firstSnapshot, secondSnapshot)
+    assertThat(returnedRelativeCpuUsage).isWithin(1e-5).of(relativeCpuUsage)
+  }
+
+  @Test
+  @RunParameterized(
+    Iteration(
+      "negativeDeltaCpuValue",
+      "secondCpuValue=900",
```

Comment:
For all the new parameterized tests: it's a bit hard to understand what's going on without seeing both the first and second values. Perhaps you could instead default both first & second snapshot values, and then override the defaults with specific parameter injections. That would let you change only the values related to the behavior under test which would probably help shorten these iterations & also make them more specific to what matters.

Reply:
per our discussion on chat, I've separated all these clubbed scenarios into separate tests.

Reply:
Thanks--I think the separation has made things much cleaner! And, based on it, I don't think any additional infrastructure work is needed at this time.

That being said, I think these tests are still missing the context that could be provided by having both the first & second values, rather than defaulting one of them.

Reply:
modified these tests to have the first and second values for added context.